students/timm67/session04/mailroom_part2.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env/python3
+
+"""
+Tim Meese
+Au2018-Py210B
+Mailroom Part 2 assignment
+"""
+
+import collections
+from collections import defaultdict
+
+#
+# Revised donors database is an array of dictionaries with appropriate keynames. 
+# 'amounts' remains a list of donation values 
+# 
+
+donors = [
+    { 'fname' : 'Jim',  'lname' : 'Tillson',  'amounts' : [5.00, 20.00]},
+    { 'fname' : 'Barb', 'lname' : 'Langley',  'amounts' : [10.00, 20.00, 100.00]},
+    { 'fname' : 'Jen',  'lname' : 'Garfield', 'amounts' : [10.00, 20.00, 5.00]},
+    { 'fname' : 'Rex',  'lname' : 'Miller',   'amounts' : [20.00, 20.00, 5.00]},
+    { 'fname' : 'Tony', 'lname' : 'Blake',    'amounts' : [20.00, 20.00, 10.00]},
+]
```

Comment:
I think the intended dict structure is something like below:
```
donors = { 'Jim Tillson': [5, 20], 'Barb Langley': [10, 20, 100], ... }
```
Or, if you want to keep first names and last names separately, something like this:
```
donors = { ('Jim', 'Tillson'): [5, 20], ('Barb', 'Langley'): [10, 20, 100], ... }
```
The key is to be able to use `if (donor_fname, donor_lname) in donors:`, without having to search for a specific entry in the list. Let me know if this doesn't make sense... Not sure if I should require this change, but definitely this is something you will want to consider.